sdks/java/core/src/main/java/org/apache/beam/sdk/io/BoundedSource.java

```diff
@@ -134,6 +136,106 @@ public Double getFractionConsumed() {
     }
 
     /**
+     * A constant to use as the return value for {@link #getParallelismConsumed()} or
+     * {@link #getParallelismRemaining()} when the exact value is unknown.
+     */
+    public static final long PARALLELISM_UNKNOWN = -1;
+
+    /**
+     * Returns the total amount of parallelism in the consumed (returned and processed) range of
+     * this reader's current {@link BoundedSource} (as would be returned by
+     * {@link #getCurrentSource}). This corresponds to all split point records (see
+     * {@link RangeTracker}) returned by this reader, <em>excluding</em> the current record.
```

Comment:
To confirm my understanding, we exclude the current record because it may not have been consumed yet?